cloud/scope/vpc_cluster.go

```diff
@@ -166,3 +197,281 @@ func (s *VPCClusterScope) Close() error {
 func (s *VPCClusterScope) Name() string {
 	return s.Cluster.Name
 }
+
+// NetworkSpec returns the VPCClusterScope's Network spec.
+func (s *VPCClusterScope) NetworkSpec() *infrav1beta2.VPCNetworkSpec {
+	return s.IBMVPCCluster.Spec.Network
+}
+
+// NetworkStatus returns the VPCClusterScope's Network status.
+func (s *VPCClusterScope) NetworkStatus() *infrav1beta2.VPCNetworkStatus {
+	return s.IBMVPCCluster.Status.Network
+}
+
+// CheckTagExists checks whether a user tag already exists.
+func (s *VPCClusterScope) CheckTagExists(tagName string) (bool, error) {
+	exists, err := s.GlobalTaggingClient.GetTagByName(tagName)
+	if err != nil {
+		return false, fmt.Errorf("failed checking for tag: %w", err)
+	}
+	return exists != nil, nil
+}
+
+// GetNetworkResourceGroupID returns the Resource Group ID for the Network Resources if it is present. Otherwise, it defaults to the cluster's Resource Group ID.
+func (s *VPCClusterScope) GetNetworkResourceGroupID() (string, error) {
+	// Check if the ID is available from Status first.
+	if s.NetworkStatus() != nil && s.NetworkStatus().ResourceGroup != nil && s.NetworkStatus().ResourceGroup.ID != "" {
+		return s.NetworkStatus().ResourceGroup.ID, nil
+	}
+
+	// Collect the Network's Resource Group Id if it is defined in Spec.
+	if s.NetworkSpec() != nil && s.NetworkSpec().ResourceGroup != nil {
```

Comment:
May be negating the if and early return would help in better formating and understanding something like.

```
if s.NetworkSpec() == nil || s.NetworkSpec().ResourceGroup == nil {
	// Otherwise, default to using the cluster's Resource Group ID.
	return s.GetResourceGroupID()
}
```

Reply:
Updated